### ISSUE TITLE
[Bug]: Datatable - Batch actions story a11y violation 

### ISSUE THREAD

**Issue by francinelucca** [carbon-design-system/carbon#13858]
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.29.0

### Description

Current [datatable batch actions story](https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default) has 1 a11y violation: "aria-labelledby attribute cannot be used on a div with no valid role attribute."
<img width="1467" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40550942/010faecb-3741-4519-b299-d433e75522d9">

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

### Steps to reproduce

https://react.carbondesignsystem.com/?path=/story/components-datatable-batch-actions--default

Go to accessibility tab and see violation information 

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Any idea why this isn't showing up as a violation through the IBM a11y checker? 

<img width="1309" alt="Screenshot 2023-05-23 at 1 07 42 PM" src="https://github.com/carbon-design-system/carbon/assets/11928039/45229492-a389-4457-b5fd-3abe19d8cee9">

**Comment by tay1orjones** [comment]
I can't repro this one either using the axe tools in the addons pane or the accessibility-checker extension



![Image](https://github.com/carbon-design-system/carbon/assets/3360588/57477c02-14af-46e6-b76e-1f9a37508c68)
